### PR TITLE
Models

### PR DESCRIPTION
**Pull request by riteshsp2000**
### Description
Create Mongoose Models for 
1. Certificates
2. Request
3. User
4. Sign

### CODE REVIEW

File: src/models/user.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    type: String,
+    required: false,
+    // We can have a default image
+  },
+  firebaseID: {
+    type: String,
+    required: true,
+  },
+  accessLevel: {
+    type: String,
```
Comment (DesignrKnight) [inline src/models/user.js:26]: make it into an array of objects

**Review by DesignrKnight — CHANGES_REQUESTED** [review]

**Comment by DesignrKnight** [comment]
Create an issue and link the same

File: src/models/request.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import mongoose from 'mongoose';
```
[the remark this answers is not in the document]
Comment (riteshsp2000) [reply src/models/request.js:1]: I tried doing that. The issue with that is the library might be having the code in old syntax which might not support the import export syntax. Importing just the functions is throwing an error

File: src/models/request.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  },
+  updatedBy: {
+    type: String,
+    required: true,
+    trim: true,
+  },
+  schemaVersion: {
+    type: Number,
+    required: false,
+    default: 1.0,
```
[the remark this answers is not in the document]
Comment (riteshsp2000) [reply src/models/request.js:117]: There is no integer data type in mongoose right. Its just Number. and generally we have versions of the format 1.x so. If only 1 is okay, cool

File: src/models/request.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    required: true,
+    default: Date.now(),
+  },
+  createdBy: {
+    type: String,
+    required: true,
+    trim: true,
+  },
+  updatedBy: {
+    type: String,
```
[the remark this answers is not in the document]
Comment (riteshsp2000) [reply src/models/request.js:110]: ObjectId makes sense, ill change it

File: src/models/user.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    default: null,
+    trim: true,
+  },
+  name: {
+    type: String,
+    required: true,
+    trim: true,
+  },
+  displayPicture: {
+    type: String,
```
[the remark this answers is not in the document]
Comment (riteshsp2000) [reply src/models/user.js:17]: yes right, changing it away

File: src/models/user.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    required: true,
+  },
+  accessLevel: {
+    type: String,
+    required: true,
+  },
+  requests: [String],
+  signs: [String],
+  certificates: [String],
```
[the remark this answers is not in the document]
Comment (riteshsp2000) [reply src/models/user.js:31]: yup ObjectId

File: src/models/user.js (hunk 1 above)
Comment (riteshsp2000) [reply src/models/user.js:26]: so as in, one user can have multiple accesslevels?

**Review by DesignrKnight — APPROVED** [review]

**Comment by DesignrKnight** [comment]
image
image

The branch has 7 commits. 5 commits are shown here. 2 vanished
Weird GitHub Issue. New PR at #38

**Pull request closed without merging**